src/pyNAVIS/plots.py

```diff
@@ -385,3 +386,323 @@ def difference_between_LR(spikes_file, settings, return_data = False, graph_tile
+
+        # Loop over data dimensions and create text annotations.
+        for i in range(len(freq_channel_labels)):
+            for j in range(len(neuron_id_labels)):
+                text = htmap_ax.text(j, i, mso_activity[i, j],
+                            ha="center", va="center", color="w", fontsize='xx-small')
+        
+        plt.title(graph_title, fontsize='x-large')
+
+        plt.tight_layout()
+        htmap_fig.show()
+
+    @staticmethod
+    def mso_spikegram(localization_file, settings, localization_settings, dot_size = 0.2, graph_tile = 'MSO spikegram', verbose = False):
+        """
+        Plots the 3D spikegram (also known as raster plot) of the MSO information contained in a LocalizationFile.
+        This is, a graph where the X axis represents neuron IDs, the Y axis means time, and the Z axis represents the frequency channels of the cochlea, and where every spike is plotted as a dot.
+
+        Parameters:
+                localization_file (LocalizationFile): Localization file to plot.
+                settings (MainSettings): Configuration parameters for the file to plot.
+                localization_settings (LocalizationSettings): Localization configuration parameters for the file to plot.
+                dot_size (float): Size of the dots used in the spikegram plot.
+                graph_tile (string, optional): Text that will appear as title for the graph.
+                verbose (boolean, optional): Set to True if you want the execution time of the function to be printed.
+
+        Returns:
+                None.
+
+        Note: 
+                None.
+        """
+
+        if verbose == True: start_time = time.time()
+
+        # REPRESENTATION
+        plt.style.use('seaborn-whitegrid')
+        msospk_fig = plt.figure()
+        ax = msospk_fig.add_subplot(111, projection='3d')
+        msospk_fig.canvas.set_window_title(graph_tile)
+
+        # Plot all the spikes stored in the localization_file
+        ax.scatter(localization_file.mso_neuron_ids, localization_file.mso_timestamps, localization_file.mso_channels, s=dot_size)
+            
+        if verbose == True: print('SPIKEGRAM CALCULATION', time.time() - start_time)
+
+        plt.title(graph_tile, fontsize='x-large')
+
+        # Set the label of each axis
+        ax.set_xlabel('Neuron ID', fontsize='large')
+        ax.set_ylabel('Timestamp ($\mu$s)', fontsize='large')
+        ax.set_zlabel('Freq. channel', fontsize='large')
+
+        # Set the axis' limits for each axis according to the LocalizationSettings parameters
+        ax.set_xlim([0, localization_settings.mso_num_neurons_channel])
+        ax.set_ylim([0, localization_file.mso_timestamps[-1]])
+        ax.set_zlim([0, settings.num_channels])
+
+        plt.tight_layout()
+        msospk_fig.show()
+
+    @staticmethod
+    def mso_localization(localization_file, settings, localization_settings, graph_title = "MSO localization estimation", start_at_zero = True, verbose = False):
```

Comment:
Change function name to mso_localization_plot --> also change it in the examples and the rest of the code!

Reply:
Done!